xds/src/main/java/io/grpc/xds/EnvoyProtoData.java

```diff
@@ -339,4 +339,270 @@ public String toString() {
           .toString();
     }
   }
+
+  static final class CidrRange {
+    private final String addressPrefix;
+    private final int prefixLen;
+
+    public CidrRange(String addressPrefix, int prefixLen) {
+      this.addressPrefix = addressPrefix;
+      this.prefixLen = prefixLen;
+    }
+
+    static CidrRange fromEnvoyProtoCidrRange(
```

Comment:
First, I used the existing pattern in `EnvoyProtoData` which seems workable.

Separating the conversion into a Converter class seems like a good idea and can be implemented when v3 protos are introduced. Also I am not sure how different/similar v3 protos will be to existing ones. There is a distinct possibility that v3 protos would force us to also update our internal xDS data types (because of new concepts) so it might not be worth our while to develop a complex conversion structure.